package/MDAnalysis/coordinates/PDB.py

```diff
@@ -1044,8 +1049,11 @@ def get_attr(attrname, default):
             vals['segID'] = segids[i][:4]
             vals['element'] = guess_atom_element(atomnames[i].strip())[:2]
 
-            # .. _ATOM: http://www.wwpdb.org/documentation/file-format-content/format32/sect9.html#ATOM
-            self.pdbfile.write(self.fmt['ATOM'].format(**vals))
+            if atom.record_type == 'ATOM':
```

Comment:
What happens here if there isn't a record type in the attributes list? We should try to capture that case and default to a sensible value (warning users that this is happening).

Reply:
This is my first contribution so I'm still learning the details of MDAnalysis. Can you provide me with the example when there is no atom.record_type? I thought that as being read from PDB with ATOM or HETATM in line, this goes into record_type? 

Is your suggestion to check for the attribute and if there is no attribute (or it is not ATOM/HETATM), we should write as `ATOM` and print a warning?

Reply:
> This is my first contribution so I'm still learning the details of MDAnalysis. Can you provide me with the example when there is no atom.record_type? I thought that as being read from PDB with ATOM or HETATM in line, this goes into record_type?

So as far as I'm aware, `record_type` is only set when reading a PDB file (maybe PDBQT, haven't checked). In most cases, say when reading a GRO/XTC combination, you won't get the `record_type` attribute (purely because the readers will only provide attributes for things they can see in the input files, and `record_type` is unique to PDB files).

> Is your suggestion to check for the attribute and if there is no attribute (or it is not ATOM/HETATM), we should write as ATOM and print a warning?

Yeah, probably the simplest case would be to put a try/except clause around this to catch a `NoDataError`, throw a warning and default to `ATOM` (which would mimic the current MDA behaviour in most cases). However, since this will happen __a lot__ this could actually lead to a big decrease in performance, so we'd have to benchmark to see how bad things get.

Reply:
Check for missing attribute added, all tests passed.